tools/src/keygen/Main.hs

```diff
@@ -106,22 +109,21 @@ getAvvmGenesis
     => AvvmStakeOptions -> m (GenesisCoreData, GenesisGtData)
 getAvvmGenesis AvvmStakeOptions {..} = do
     jsonfile <- liftIO $ BSL.readFile asoJsonPath
-    holder <- getHolderId asoHolderKeyfile
     case eitherDecode jsonfile of
         Left err       -> error $ toText err
         Right avvmData -> do
             avvmDataFiltered <- liftIO $ applyBlacklisted asoBlacklisted avvmData
             let totalAvvmStake = sum $ map aeCoin $ utxo avvmDataFiltered
             putText $ "Total avvm stake after applying blacklist: " <> show totalAvvmStake
-            pure $ genGenesis avvmDataFiltered asoIsRandcerts holder
+            -- warning: if testnet is not enabled,
+            -- gcdBootstrapAddresses will be empty. We should come up
+            -- with a way to pass real boot addresses to avvm
+            -- distribution.
+            pure $ genGenesis avvmDataFiltered asoIsRandcerts mempty
```

Comment:
You always pass `mempty` as the last argument of `genGenesis`. This `mempty` is passed to `genesisSplitBoot`. It returns empty distribution. It's leads to wrong distribution and I don't see a place where it's fixed.

Regarding CSL-1308: I already realized it while I was reading this comment and it was clear why having empty `gcdBootstrapStakeholders` is not big problem now. But having empty distributions in `ExplicitStakes` seems totally wrong.